src/plugins/intel_gpu/src/graph/impls/ocl/rope.cpp

```diff
@@ -46,15 +46,50 @@ struct rope_impl : typed_primitive_impl_ocl<rope> {
         params.axis = primitive->config.is_qwen || primitive->config.is_chatglm ? 2 : 3;
         params.num_of_inputs = primitive->config.is_chatglm || primitive->config.is_interleaved ? 2 : 3;
 
+        if (primitive->gather_rank > 0) {
+            params.gather_rank = primitive->gather_rank;
+            params.num_of_inputs++;
```

Comment:
Does this `num_of_inputs` parameter differ from `impl_param.input_layouts.size()` ?